### PR TITLE
DEP: update minimal recommendation for matplotlib (3.3.4 -> 3.6.0)

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
matplotlib 3.5.0 was the first version to support Python 3.10 so we're de facto not supporting 3.3 and 3.4 anymore

This originated from https://github.com/conda-forge/astropy-feedstock/pull/137#discussion_r1635029156

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
@eerovaher reminded me that image tests were still somehow setup to use mpl 3.3.4 so I'm going to update that too.

FYI @cadair

**Comment by neutrinoceros** [comment]
switching to draft for now, until I know for sure that CI stays green.

**Comment by neutrinoceros** [comment]
of course one blocker for such PR is that the circleci job named mpl334 is currently required for merging

**Comment by pllim** [comment]
1. In PR like this, I can override the branch protection checks because I will have to update the rules after merge, so don't sweat about mpl334 missing. :)
2. Should we delete `py310-test-image-mpl334-cov.json`?
3. By [SPEC 0](https://scientific-python.org/specs/spec-0000/), should we bump straight to matplotlib 3.6? 🤔

**Comment by Cadair** [comment]
SPEC 0 says bump to 3.6.0 in Q3 2024 which would be our next release right? So yeah I think we should bump to 3.6.0 now.

**Comment by astrofrog** [comment]
If we need to bump to 3.6.0 now we should rename the mpl350 JSON file to mpl360, add a mpl360 job and see if any of the hashes need to be updated.

**Comment by neutrinoceros** [comment]
By popular demand, now bumping to 3.6.0 !

**Comment by pllim** [comment]
LGTM but needs a change log. Thanks!

**Comment by neutrinoceros** [comment]
done !

**Review by pllim — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -50,7 +50,7 @@ dependencies = [
 # Recommended run-time dependencies to enable a lot of functionality within Astropy.
 recommended = [
     "scipy>=1.8",
-    "matplotlib>=3.3,!=3.4.0,!=3.5.2",
+    "matplotlib>=3.5.0,!=3.5.2",
```
Comment (neutrinoceros) [inline pyproject.toml:53]: Here's what I get for coding on a train

```suggestion
    "matplotlib>=3.6.0",
```

**Comment by pllim** [comment]
Updated the branch protection rules. This means all open PRs will need a rebase. FYI.

Thanks!

**Pull request merged**